blockchain/miner.py

```diff
@@ -39,8 +43,15 @@ def valid_proof(last_hash, proof):
     IE:  last_hash: ...AE9123456, new hash 123456E88...
     """
 
-    # TODO: Your code here!
-    pass
+    # Check if hash is valid by comparing
+    # last 6 chars of last_hash to first 6 chars of new proof
+    # If they match, proof valid
+
+    check_proof_hash = hashlib.sha256(f'{proof}'.encode()).hexdigest()
+    check_proof_hash_beg_check = check_proof_hash[:6]
+    last_hash_end_check = last_hash[-6:]
+
+    return check_proof_hash_beg_check == last_hash_end_check
 
 
 if __name__ == '__main__':
```

Comment:
Did you have trouble accessing the test server?  I know some folks did. I think it's because there is no `/` at the end of the URL.

Reply:
Ope - never mind I see it's added down below.